Add login email support for wildmeta

silva-fj opened the pull request:
## Summary

- Added new RPC method `omni_getApproveAgentWalletData` to support email-based authentication for wildmeta agent wallet approvals
- Generates EIP-712 signatures for agent approval on Hyperliquid network

silva-fj committed:
Add RPC method to approve agent wallet data for wildmeta

Implements omni_getApproveAgentWalletData endpoint that:
- Verifies email authentication
- Derives main wallet address from email identity
- Generates EIP-712 signature for agent approval on Hyperliquid

silva-fj committed:
refactor(rpc): remove redundant comments in EIP712 signature fn

Kailai-Wang commented on tee-worker/omni-executor/rpc-server/src/methods/omni/get_approve_agent_wallet_data.rs:
```diff
@@ -0,0 +1,337 @@
+	pub client_id: String,
+	pub agent_address: String,
+	pub agent_name: Option<String>,
+	pub chain_id: ChainId,
+	pub is_testnet: bool,
```
Hmm - isn't `chain_id` enough to tell it's Testnet or Mainnet?

silva-fj replied:
Yes, we just need to map it. I'll update this

Kailai-Wang commented on tee-worker/omni-executor/rpc-server/src/methods/omni/get_approve_agent_wallet_data.rs:
```diff
@@ -0,0 +1,337 @@
+					ErrorObject::from(ErrorCode::ServerError(AUTH_VERIFICATION_FAILED_CODE))
+				})?;
+
+			let identity = Identity::from_web2_account(&email, Web2IdentityType::Email);
+			let omni_account = identity.to_omni_account(&params.client_id);
```
I recommend we use `UserId` -> `Identity` conversion directly: it's more actively maintained.

Kailai-Wang reviewed (approved):
Looks good to me - I'm purely reviewing it from the code logic itself tho, the concrete context (if it meets requirements) needs to be checked by @BillyWooo

silva-fj committed:
refactor(rpc): improve wildmeta agent wallet approval implementation

- Remove redundant is_testnet parameter, determine from chain ID directly
- Use direct UserId to Identity conversion (more actively maintained)
- Add is_testnet_chain helper function with known testnet/mainnet IDs

Kailai-Wang committed:
Merge branch 'dev' into p-1642-add-login-email-support-for-wildmeta

BillyWooo reviewed (approved):
Thanks. LGTM
